app/controllers/alchemy/api/nodes_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class Api::NodesController < Api::BaseController
+    before_action :load_node
+
+    def move
+      target_parent_node = Node.find(params[:target_parent_id])
+      @node.move_to_child_with_index(target_parent_node, params[:new_position])
+      render json: @node, serializer: NodeSerializer
+    end
+
+    def toggle_folded
+      node = Node.find(params[:id])
+      node.update(folded: !node.folded)
+      render json: @node, serializer: NodeSerializer
```

Comment:
The same here. We need authorization and we already loaded `@node` so we do not need to load it twice.

```suggestion
      authorize! :update, @node
      @node.update(folded: !@node.folded)
      render json: @node, serializer: NodeSerializer
```